src/services/filter/expression/mod.rs

```diff
@@ -1,6 +1,6 @@
 //! Shared logic for parsing and evaluating expressions
 
-mod ast;
+pub mod ast;
```

Comment:
Seems odd that this is the only one marked as public? I reckon you could import this from the filter mod instead?